src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -0,0 +1,76 @@
+import json
+
+import d3a.constants
+from d3a.d3a_core.redis_connections.redis_area_market_communicator import RedisCommunicator
+from d3a.models.market import validate_authentic_bid_offer_pair
+from d3a.models.market.market_structures import BidOfferMatch
+from d3a.models.myco_matcher.base_matcher import BaseMatcher
+
+
+class ExternalMatcher(BaseMatcher):
+    def __init__(self):
+        super(ExternalMatcher, self).__init__()
+        self.job_uuid = d3a.constants.COLLABORATION_ID
+        self.myco_ext_conn = None
+        self.channel_prefix = f"external-myco/{self.job_uuid}/"
+        self._setup_redis_connection()
+        self.area_uuid_markets_mapping = {}
+        self.markets_mapping = {}
+        self.recommendations = []
+
+    def _setup_redis_connection(self):
+        self.myco_ext_conn = RedisCommunicator()
+        self.myco_ext_conn.sub_to_channel("external-myco/get_job_uuid",
+                                          self.get_job_uuid)
+        self.myco_ext_conn.sub_to_channel(f"{self.channel_prefix}get_offers_bids/",
+                                          self.publish_offers_bids)
+        self.myco_ext_conn.sub_to_channel(f"{self.channel_prefix}post_recommendations/",
+                                          self.match_recommendations)
+
+    def publish_offers_bids(self, message):
+        """
+        Function that queries publishes open offers and bids
+        published data are of the following format
+        market_offers_bids_list_mapping = {"market_id" : {"bids": [], "offers": [] }, }
+        """
+        # TODO: message can contain filters
+        data = {"event": "offers_bids_response"}
+        market_offers_bids_list_mapping = {}
+        for area_uuid, markets in self.area_uuid_markets_mapping.items():
```

Comment:
You are not using the `area_uuid` here. I dont see a danger of hash collisions, so the market_ids should be enough to distinguish between the markets, right?

Reply:
Any opinion here @Hassan754 . If you are not using the area_uuid, please use `.values()` instead. Thanks

Reply:
This still needs to be dealt with.

Reply:
@hannesdiedrich in the Test 2 we need to pass the area_uuid to the Myco client, this is why I left it here

Reply:
OK! Understood.